addons/website_event/models/event_event.py

```diff
@@ -413,15 +413,16 @@ def _create_menu(self, sequence, name, url, xml_id, menu_type):
         """
         self.browse().check_access('write')
         view_id = False
-        if not url:
+        if (not url) or (url and xml_id):
```

Comment:
So now we'll be creating a whole view that copies the template, only to store seo data and never use the view..

Seems a bit wasteful actually. I would say we either always use the view, i.e. each website has its own track proposals design (does not seem like such a crazy concept). Or we let `website.event.menu` hold the SEO data by inheriting `website.seo.metadata`, which seems more appropriate for the scope of the task.

Reply:
Hello @reth-odoo 

I understand the part of inheriting `website.seo.metadata` in `website.event.menu`, but the pages shown are the record of `event.event` thus making an extra view for the seo data. So, how will `website.event.menu` helps here? What am I missing here.

Thanks :smile:

Reply:
@nihp-odoo Basically right now you're only using the views to store SEO data. Functionally we don't care about creating the views, because we never use the contents of the view, we only use them to pass an `seo_object` to the front-end. Then the website uses that `seo_object` as the place to store and fetch SEO data about the page that's rendered.

So instead of using the view as a storage medium, I'm saying we use the `website.event.menu` that currently holds the view. As *that* is unique to each page of each event, without having to change anything.

Instead of `render_values['seo_object'] = event.track_menu_ids.view_id.website_meta_title` you would just do `render_values['seo_object'] = even.track_menu_ids.website_meta_title`. And whether there's a custom view for the menu or not will be irrelevant, we'll just store the title on the event menu.

Reply:
Hello @reth-odoo 

Okay! I got your point on using `website.event.menu`. But when we save SEO data, it is stored in the model whose record we are using for view. (According to the current flow)
Can you please elaborate on how to achieve it using `website.event.menu`.

Thanks :blush:

Reply:
Indeed the record will be the same for all websites, but anyway the page is also the same for all events on one website so I don't think it's a big deal. If we wanted fine-grained control all the way down we'd just use the views every time. I'll check with jija just in case though.

And anyway it's the same issue as using "event" as the seo storage, so the register page has the same issue (which is fine).

>  But when we save SEO data, it is stored in the model whose record we are using for view

It only does that if you don't pass "seo_object" in your template, that key was added specifically for this use case.
If you pass a record as "seo_object" when rendering the main website template, it will use that record to store SEO data. That's why it's using the views instead of the event.

So in all the routes where we add `seo_object`, you just need to pass any object that inherits `website.seo.metadata`.

- make `website.event.menu` inherit `website.seo.metadata`
- set `render_values['seo_object'] = event.introduction_menu_ids` in all the specific routes, instead of the setting that to be the view currently
- done (hopefully)

Just tried it and it seems to work fine.
As I said I'll check with jija so no need to make the change just yet, will get back to you.